framework-cloud/framework-openshift/src/main/java/org/kie/cloud/openshift/scenario/OpenShiftScenario.java

```diff
@@ -154,4 +169,22 @@ protected void logNodeNameOfAllInstances() {
     public void addDeploymentScenarioListener(DeploymentScenarioListener<T> deploymentScenarioListener) {
         deploymentScenarioListeners.add(deploymentScenarioListener);
     }
+
+    public void addExtraDeployment(ExternalDeployment<?, ?> externalDeployment) {
```

Comment:
Not sure if this is the best approach, considering that different deployments could have different behaviour. For example if we want to start Kie server with already predefined kjar deployments we need to have Nexus already deployed and running (with needed kjars deployed). In case we will deploy kjars to Nexus in tests then it is ok to start Nexus in parallel with other deployments.

Reply:
You mean if the nexus needs to be completely deployed before starting the rest ?
Then we should consider waitingforscale and put it as a parameter of the method addExtraDeployment or ?

Reply:
Sounds ok, or better to create another method to handle this, something like addExtraDeploymentSynchronized with javadoc explaining that product deployment is triggered when this external deployment is fully started.

Reply:
right, would make sense to have 2 different methods for 2 different behavior indeed :)

Reply:
made a modification. Let me know if you are ok with that